### DISCUSSION TITLE
Apple Missing privacy manifest for GoogleSignIn & GTMAppAuth

### DISCUSSION THREAD

**Discussion by br0ll** [capawesome-team/capacitor-firebase#736] [Q&A] [answered]
Hey,

When submitting our app for Apple App Review, the following warnings show up in the review process, indicating there needs to ba taken an action before Nov 11 or we won't be able to submit app updates.
We have made our own Privacy Manifest for the app itself, but I can't find any info on how to do the same for external frameworks, specifically the ones required by the FirebaseAuthentication package.
Do you have any experience with this?

> ITMS-91061: Missing privacy manifest - Your app includes “Frameworks/GTMAppAuth.framework/GTMAppAuth”, which includes GTMAppAuth, an SDK that was identified in the documentation as a privacy-impacting third-party SDK. Starting November 12, 2024, if a new app includes a privacy-impacting SDK, or an app update adds a new privacy-impacting SDK, the SDK must include a privacy manifest file. Please contact the provider of the SDK that includes this file to get an updated SDK version with a privacy manifest. For more details about this policy, including a list of SDKs that are required to include signatures and manifests, visit: https://developer.apple.com/support/third-party-SDK-requirements.
> 
> ITMS-91061: Missing privacy manifest - Your app includes “Frameworks/GoogleSignIn.framework/GoogleSignIn”, which includes GoogleSignIn, an SDK that was identified in the documentation as a privacy-impacting third-party SDK. Starting November 12, 2024, if a new app includes a privacy-impacting SDK, or an app update adds a new privacy-impacting SDK, the SDK must include a privacy manifest file. Please contact the provider of the SDK that includes this file to get an updated SDK version with a privacy manifest. For more details about this policy, including a list of SDKs that are required to include signatures and manifests, visit: https://developer.apple.com/support/third-party-SDK-requirements.
> 
>

**Comment by robingenz** [comment]
You only need to update to the latest version of the plugins. The privacy manifest should already be included there.

**Reply by br0ll** [reply]
Ah, of course. I was running a local fork (re https://github.com/capawesome-team/capacitor-firebase/pull/711), but I've merged in the latest version now!

**Reply by curlyz** [reply]
I'm still having problem with @capacitor-firebase 6.1.0
I have also try to copy the privacy manually like this but no help
```
<?xml version="1.0" encoding="UTF-8"?>
<!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
<plist version="1.0">
  <dict>
    <key>NSPrivacyTracking</key>
    <false/>
    <key>NSPrivacyTrackingDomains</key>
    <array/>
    <key>NSPrivacyCollectedDataTypes</key>
    <array>
      <dict>
        <key>NSPrivacyCollectedDataType</key>
        <string>NSPrivacyCollectedDataTypeName</string>
        <key>NSPrivacyCollectedDataTypeLinked</key>
        <true/>
        <key>NSPrivacyCollectedDataTypeTracking</key>
        <false/>
        <key>NSPrivacyCollectedDataTypePurposes</key>
        <array>
          <string>NSPrivacyCollectedDataTypePurposeAppFunctionality</string>
        </array>
      </dict>
      <dict>
        <key>NSPrivacyCollectedDataType</key>
        <string>NSPrivacyCollectedDataTypeEmailAddress</string>
        <key>NSPrivacyCollectedDataTypeLinked</key>
        <true/>
        <key>NSPrivacyCollectedDataTypeTracking</key>
        <false/>
        <key>NSPrivacyCollectedDataTypePurposes</key>
        <array>
          <string>NSPrivacyCollectedDataTypePurposeAppFunctionality</string>
        </array>
      </dict>
      <dict>
        <key>NSPrivacyCollectedDataType</key>
        <string>NSPrivacyCollectedDataTypePhoneNumber</string>
        <key>NSPrivacyCollectedDataTypeLinked</key>
        <true/>
        <key>NSPrivacyCollectedDataTypeTracking</key>
        <false/>
        <key>NSPrivacyCollectedDataTypePurposes</key>
        <array>
          <string>NSPrivacyCollectedDataTypePurposeAppFunctionality</string>
        </array>
      </dict>
      <dict>
        <key>NSPrivacyCollectedDataType</key>
        <string>NSPrivacyCollectedDataTypeOtherDataTypes</string>
        <key>NSPrivacyCollectedDataTypeLinked</key>
        <true/>
        <key>NSPrivacyCollectedDataTypeTracking</key>
        <false/>
        <key>NSPrivacyCollectedDataTypePurposes</key>
        <array>
          <string>NSPrivacyCollectedDataTypePurposeAppFunctionality</string>
          <string>NSPrivacyCollectedDataTypePurposeAnalytics</string>
        </array>
      </dict>
      <dict>
        <key>NSPrivacyCollectedDataType</key>
        <string>NSPrivacyCollectedDataTypeCoarseLocation</string>
        <key>NSPrivacyCollectedDataTypeLinked</key>
        <true/>
        <key>NSPrivacyCollectedDataTypeTracking</key>
        <false/>
        <key>NSPrivacyCollectedDataTypePurposes</key>
        <array>
          <string>NSPrivacyCollectedDataTypePurposeAppFunctionality</string>
        </array>
      </dict>
      <dict>
        <key>NSPrivacyCollectedDataType</key>
        <string>NSPrivacyCollectedDataTypeUserID</string>
        <key>NSPrivacyCollectedDataTypeLinked</key>
        <true/>
        <key>NSPrivacyCollectedDataTypeTracking</key>
        <false/>
        <key>NSPrivacyCollectedDataTypePurposes</key>
        <array>
          <string>NSPrivacyCollectedDataTypePurposeAppFunctionality</string>
          <string>NSPrivacyCollectedDataTypePurposeAnalytics</string>
        </array>
      </dict>
      <dict>
        <key>NSPrivacyCollectedDataType</key>
        <string>NSPrivacyCollectedDataTypeDeviceID</string>
        <key>NSPrivacyCollectedDataTypeLinked</key>
        <true/>
        <key>NSPrivacyCollectedDataTypeTracking</key>
        <false/>
        <key>NSPrivacyCollectedDataTypePurposes</key>
        <array>
          <string>NSPrivacyCollectedDataTypePurposeAnalytics</string>
        </array>
      </dict>
      <dict>
        <key>NSPrivacyCollectedDataType</key>
        <string>NSPrivacyCollectedDataTypeOtherUsageData</string>
        <key>NSPrivacyCollectedDataTypeLinked</key>
        <true/>
        <key>NSPrivacyCollectedDataTypeTracking</key>
        <false/>
        <key>NSPrivacyCollectedDataTypePurposes</key>
        <array>
          <string>NSPrivacyCollectedDataTypePurposeAnalytics</string>
        </array>
      </dict>
    </array>
    <key>NSPrivacyAccessedAPITypes</key>
    <array>
      <dict>
        <key>NSPrivacyAccessedAPITypeReasons</key>
        <array>
          <string>CA92.1</string>
        </array>
        <key>NSPrivacyAccessedAPIType</key>
        <string>NSPrivacyAccessedAPICategoryUserDefaults</string>
      </dict>
      <dict>
        <key>NSPrivacyAccessedAPIType</key>
        <string>NSPrivacyAccessedAPICategoryFileTimestamp</string>
        <key>NSPrivacyAccessedAPITypeReasons</key>
        <array>
          <string>C617.1</string>
        </array>
      </dict>
    </array>
  </dict>
</plist>

```

and this is my package.json
```
    "@capacitor-community/bluetooth-le": "6.0.1",
    "@capacitor-firebase/authentication": "6.1.0",
    "@capacitor/android": "6.1.2",
    "@capacitor/app": "6.0.1",
    "@capacitor/browser": "6.0.3",
    "@capacitor/core": "6.1.2",
    "@capacitor/geolocation": "^6.0.1",
    "@capacitor/haptics": "6.0.1",
    "@capacitor/ios": "6.1.2",
    "@capacitor/keyboard": "6.0.2",
    "@capacitor/local-notifications": "6.1.0",
    "@capacitor/motion": "^6.0.1",
    "@capacitor/screen-orientation": "6.0.2",
    "@capacitor/status-bar": "6.0.1",
    "@capacitor/toast": "6.0.2",
    "@capawesome/capacitor-screen-orientation": "^6.0.1",
```

**Reply by vlafranca** [reply]
Also having this issue with plugin version 6.1.0 is there something I missed ?

**Reply by robingenz** [reply, marked as answer]
I updated the Firebase iOS SDK to the latest version a few days ago. This is the pre release: https://github.com/capawesome-team/capacitor-firebase/runs/31658521770 Feel free to give it another try.

**Reply by curlyz** [reply]
Thanks @robingenz , it works.
Apple no longer sent complain email.
Hope that it will be release soon.

**Reply by robingenz** [reply]
I've just published the release: https://github.com/capawesome-team/capacitor-firebase/pull/702

**Reply by vlafranca** [reply]
Thanks !

**Comment by vlafranca** [comment]
Just got this error when updating to version 6.2.0, seems there is a conflict with the messaging plugin: 

`


Show command
✔ Copying web assets from dist to android/app/src/main/assets/public in 39.60ms
✔ Creating capacitor.config.json in android/app/src/main/assets in 516.83μs
✔ copy android in 78.11ms
✔ Copying web assets from dist to ios/App/App/public in 39.37ms
✔ Creating capacitor.config.json in ios/App/App in 164.29μs
✔ copy ios in 88.00ms
✔ copy web in 9.40ms
[warn] The bundledWebRuntime configuration option has been deprecated. Can be safely deleted.
✔ Updating iOS plugins in 5.50ms
✖ Updating iOS native dependencies with pod install - failed!
✖ update ios - failed!
[error] Analyzing dependencies
      Adding spec repo `trunk` with CDN `https://cdn.cocoapods.org/`
      [!] CocoaPods could not find compatible versions for pod "GoogleDataTransport":
      In Podfile:
      CapacitorFirebaseMessaging (from `../../node_modules/@capacitor-firebase/messaging`) was resolved to 6.2.0, which depends on
      FirebaseMessaging (\~> 11.0) was resolved to 11.4.0, which depends on
      GoogleDataTransport (\~> 10.0)
      CapacitorMlkitBarcodeScanning (from `../../node_modules/@capacitor-mlkit/barcode-scanning`) was resolved to 6.2.0, which depends on
      GoogleMLKit/BarcodeScanning (= 5.0.0) was resolved to 5.0.0, which depends on
      GoogleMLKit/MLKitCore (= 5.0.0) was resolved to 5.0.0, which depends on
      MLKitCommon (~> 10.0.0) was resolved to 10.0.0, which depends on
      GoogleDataTransport (~> 9.0)

**Reply by curlyz** [reply]
You need to delete Pod.lock

**Comment by vlafranca** [comment]
Already did but still the same, seems to come from
capacitor-mlkit/barcode-scanning plugin in fact :(

**Reply by robingenz** [reply]
Do you use the `next` version, see [Installation](https://github.com/capawesome-team/capacitor-mlkit/tree/next/packages/barcode-scanning#installation)?

**Reply by vlafranca** [reply]
Yes I do

**Reply by vlafranca** [reply]
I opened an issue on their repo (https://github.com/capawesome-team/capacitor-mlkit/issues/201), and also created a repro project if you are curious about it: https://github.com/vlafranca/capacitor-repro
